tests/dofs/distribute_dofs_01.cc

```diff
@@ -0,0 +1,55 @@
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_nedelec.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include "../tests.h"
+
+// Check DoFHandler::distribute_dofs with multiple calls to reinit
+
+void
+test()
+{
+  dealii::Triangulation<3> tria;
+  dealii::GridGenerator::hyper_cube(tria, -1, 1);
+
+  dealii::FE_Nedelec<3> fe_nedelec(0);
+
+  dealii::DoFHandler<3> dh_nedelec;
+
+  dh_nedelec.reinit(tria);
+  dh_nedelec.distribute_dofs(fe_nedelec);
+
+  dh_nedelec.reinit(tria);
+  dh_nedelec.distribute_dofs(fe_nedelec);
+
+  try
+    {
+      for (auto cell = dh_nedelec.begin_active(); cell != dh_nedelec.end();
+           ++cell)
+        {
+          for (unsigned int i = 0; i < dealii::GeometryInfo<3>::lines_per_cell;
+               ++i)
+            {
+              auto line_index = cell->line(i)->dof_index(0);
+              (void)line_index;
+            }
+        }
+
+      deallog << "ok" << std::endl;
+    }
+  catch (...)
+    {}
```

Comment:
Why are you ignoring any exception thrown here? I would just remove the whole try-catch block.

Reply:
The intention I use try-catch block is to avoid the test run into assertion failure. If the test passes, it will print `ok`, otherwise nothing will be printed. Does this make sense?

Reply:
`ctest` handles this automatically for you: if an assertion is triggered, the test will be considered as `failed`. You can safely remove the `try/catch` block here.